ogusa_calibrate/household_structure.py

```diff
@@ -0,0 +1,146 @@
+
+
+import pickle
+import pandas as pd
+import numpy as np
+import microdf as mdf
+import matplotlib.pyplot as plt
+from scipy.stats import kde
+
+panel_li = pickle.load(open('psid_lifetime_income.pkl', 'rb')) #created by psid_data_setup.py
+
+panel_li.insert(len(panel_li.columns),"weight",1) #create column of only 1s which is used as weights for taking microdf average
+panel_li.insert(len(panel_li.columns),"nu18",0)
+panel_li.insert(len(panel_li.columns),"n1864",0)
+panel_li.insert(len(panel_li.columns),"n65",0)
+#RData file and therefore dataset in psid_data_setup.py -> panel_li is assumed to have 'num_family' as follows the updated list of variables pulled
+
+def add65(head_age,spouse_age):
+  count = 0
+  if(head_age >= 65):
+    count += 1
+  if(spouse_age>=65):
+    count += 1
+  return count #assumes only head or spouse of head can be 65+
+panel_li['n65'] = panel_li.apply(lambda x: add65(x['head_age'],x['spouse_age']), axis=1)
+
+def add1864(head_age,spouse_age,num_family,num_18):
+  count = num_family-num_18
+  if(head_age >= 65):
+    count -= 1
+  if(spouse_age>=65):
+    count -= 1
+  return count
+panel_li['n1864'] = panel_li.apply(lambda x: add1864(x['head_age'],x['spouse_age'],x['num_family'],x['num_children_under18']), axis=1)
+
+panel_li['nu18'] = panel_li.apply(lambda x: x['num_children_under18'], axis=1) #assumes only children can be <18
+
+panel_li2 = panel_li.reset_index()
+panel_li3 = panel_li2[panel_li2['head_age'] <= 80]
+panel_li3 = panel_li3[panel_li3['head_age'] >= 20]
+panel_li4 = panel_li3.groupby(['head_age', 'li_group']).apply(
```

Comment:
```suggestion
panel_li_group = panel_li_20_80.groupby(['head_age', 'li_group'])[["nu18", "n1864", "n65"]].mean()
```
This takes care of much of the below code. Don't need `microdf` since nothing's weighted.